[WIP]: Adding real-time editor dashboard

arfon opened the pull request:
This PR adds additional functionality to the editor dashboard. Significantly, **this dashboard is live**, that is, it should not be necessary to use GitHub notifications to manage your papers as a JOSS editor once this goes live.

This provides three new views in addition to the existing dashboard:

- `http://joss.theoj.org/dashboard/incoming` - A list of the current papers without an editor
- `http://joss.theoj.org/dashboard/arfon` - (or your editor login) - A list of all of the papers you are currently assigned to as editor to
- `http://joss.theoj.org/dashboard/all` - A list of all papers in JOSS (excluding rejected or withdrawn submissions)
- `http://joss.theoj.org/dashboard/stats` - The JOSS stats page (currently this is at `http://joss.theoj.org/dashboard)`

@openjournals/joss-editors - Please take a look at the quick screencast below and let me know if there's information you'd like to see added to these views. One obvious extension for example would be to colour-code the rows based on some logic about how long it has been since anyone has been active on the review.

![dashboard](https://user-images.githubusercontent.com/4483/46384719-d158ae00-c685-11e8-9945-f8b52c9e6d74.gif)

**Still to be done**
- [ ] Run the initial ingest script on the production database
- [ ] Validate the origin of the incoming webhooks from GitHub

katyhuff commented:
Neato!

It might be nice (perhaps I missed it) to see the submission date (or, more stressful, days since submitted) in one of the columns. Time really moves fast and it might be motivating to be reminded of which papers have been patiently waiting for attention.

danielskatz commented:
For papers under review, can the completed/total checkbox fraction be included (e.g, 0/18, 33/36)?
And will last activity pick up that someone has checked a box, or does it just pick up new comments?

danielskatz commented:
second column title (Reviews) is misleading - if this could be longer, I would like to add the word "issues", maybe to "pre/review issues"

danielskatz commented:
> second column title (Reviews) is misleading - if this could be longer, I would like to add the word "issues", maybe to "pre/review issues"

Actually, thinking about this a bit more, I would suggest just "review issue" and point to the review issue if it exists, and the pre-review issue if there is no review issue yet.  You can always go from review to pre-review with one extra click, so I think clarity here in pointing to the one thing that's most useful (the current issue) would be helpful.

Also, **this looks really good!**

danielskatz commented:
The dashboard has pages of 10 papers - could this 10 be use settable, so for example, I could see all the papers with no editor in a single view?

danielskatz commented:
Ok, here's an interesting one:

When I go to the stats page, the second part is Accepted papers by month by editor.  There's a pulldown to select an editor, and that defaults to me.  However, the graph defaults to @arfon.  If I really want to see me, I have to change to someone else (even Arfon) then change back to me.

So, either the pulldown should start as arfon, or the graph should be built for me - in either case, they should start off being consistent.

danielskatz commented:
Can we add a link to (editor dashboard) from the top of the main JOSS page?  If this makes sense, given that some viewers might not be able to use the link...

labarba commented:
Responding to @danielskatz — It's bad form to add a visible link that leads to nowhere for most users.

danielskatz commented:
I disagree, if it's clearly marked, for example, as for editors.

danielskatz commented:
since I'm having difficulty knowing what parts of the comments have been handled and what have not, I will repeat:

> When I go to the stats page, the second part is Accepted papers by month by editor. There's a pulldown to select an editor, and that defaults to me. However, the graph defaults to @arfon. If I really want to see me, I have to change to someone else (even Arfon) then change back to me.
> 
> So, either the pulldown should start as arfon, or the graph should be built for me - in either case, they should start off being consistent.

arfon commented:
Thanks for the feedback folks. Apologies for being slow to reply. I'll attempt to 

@katyhuff said:

> It might be nice (perhaps I missed it) to see the submission date (or, more stressful, days since submitted) in one of the columns. Time really moves fast and it might be motivating to be reminded of which papers have been patiently waiting for attention.

I've added this to the dashboard view.

@danielskatz said:

> For papers under review, can the completed/total checkbox fraction be included (e.g, 0/18, 33/36)?
And will last activity pick up that someone has checked a box, or does it just pick up new comments?

This is hard to do unfortunately. The dashboard currently picks up when a checklist has been edited (or any edits have been made) but I've not figured out how to display how 'complete' the checklist is:

<img width="1280" alt="screen_shot_2018-10-17_at_4_41_07_pm" src="https://user-images.githubusercontent.com/4483/47115136-a0eb4500-d22b-11e8-8e02-199c1a24e864.png">

@danielskatz said:
> Can we add a link to (editor dashboard) from the top of the main JOSS page? If this makes sense, given that some viewers might not be able to use the link...

Yes, I've just added it. It's only visible to editors (you therefore have to be logged in to see the link)

<img width="1032" alt="editors" src="https://user-images.githubusercontent.com/4483/47115512-c298fc00-d22c-11e8-89ee-421430f0e0be.png">

@danielskatz said:

> Actually, thinking about this a bit more, I would suggest just "review issue" and point to the review issue if it exists, and the pre-review issue if there is no review issue yet. You can always go from review to pre-review with one extra click, so I think clarity here in pointing to the one thing that's most useful (the current issue) would be helpful.

Yeah, I think this needs some reworking. I currently don't like what we have. 

> For papers that are under review, I wonder if the first column link should point to the issue, rather than to the JOSS page for the paper?

Agreed. Not yet implemented.

> When I go to the stats page, the second part is Accepted papers by month by editor. There's a pulldown to select an editor, and that defaults to me. However, the graph defaults to @arfon. If I really want to see me, I have to change to someone else (even Arfon) then change back to me.

> So, either the pulldown should start as arfon, or the graph should be built for me - in either case, they should start off being consistent.

Yep, I should fix that too. Will investigate a better approach here.

> The dashboard has pages of 10 papers - could this 10 be user-settable, so for example, I could see all the papers with no editor in a single view?

It could be, but it would be quite a lot of work. Could we just agree on a number we can agree to to paginate on (e.g. 20)?

> In papers with no editor, some papers have the paused label. Maybe papers with no editor should have a toggle so that it can either show all papers with no editor or just the ones that are not paused?

Yes, I agree this is confusing. I'll think on what might be a better approach here.

Thanks for the feedback thus far everyone! ✨
